libs/wingc/src/type_check.rs

```diff
@@ -1440,7 +1471,27 @@ impl<'a> TypeChecker<'a> {
 
 				// Make sure this is a function signature type
 				let func_sig = if let Some(func_sig) = func_type.as_function_sig() {
-					func_sig
+					func_sig.clone()
+				} else if let Some(res) = func_type.as_resource() {
+					// return the signature of the "handle" method
+					let lookup_res = res.get_env().lookup_ext(&HANDLE_METHOD_NAME.into(), None);
```

Comment:
might be able to make this cleaner with `res.get_method(...)`

Reply:
Done